packages/server/lib/browsers/chrome.ts

```diff
@@ -329,10 +328,123 @@ const _handleDownloads = async function (client, dir, automation) {
   })
 }
 
-const _setAutomation = (client, automation) => {
-  return automation.use(
-    new CdpAutomation(client.send, client.on, automation),
-  )
+let frameTree
+let gettingFrameTree
+
+// eslint-disable-next-line @cypress/dev/arrow-body-multiline-braces
+const _updateFrameTree = (client) => async () => {
+  debug('update frame tree')
+
+  gettingFrameTree = new Promise<void>(async (resolve) => {
```

Comment:
Asyncs inside of a `promise` or `then` are kind of a red flag for me. I believe that anything thrown from 339 will be swallowed. Wondering if we should do something like this instead:

```
  gettingFrameTree = client.send('Page.getFrameTree').then((tree) => {
    debug('frame tree updated')
    frameTree = tree
    gettingFrameTree = null
  })
```

Reply:
I had to [revert this](https://github.com/cypress-io/cypress/pull/21137/commits/d96baa553c809aa15d1f8a87fd9341be9de69349) as it was causing all the multi-domain tests to fail. I believe the reason is that CDP commands cannot be sent synchronously from inside a CDP event handler. That's the reason we can't call `Page.getFrameTree` on demand when we need it inside of the `Fetch.requestPaused` handler. So it needs to be async and whatever way that's accomplished, it will have to be wrapped in a promise to ensure the tree is not stale when used in the `Fetch.requestPaused` handler.

I'll add some error handling in case `client.send('Page.getFrameTree')` throws an error, but I think the best we can do in that case is debug log the error.

Reply:
Added error handling in https://github.com/cypress-io/cypress/pull/21137/commits/11d63309475be5d94ec8a3c45fc5b001b9199ea1